### PR TITLE
xds: fix init fetch timeout stat

### PR DESCRIPTION
**Pull request by ramaraochavali**
Commit Message: Fixes init fetch timeout stat increment
Additional Description: fix init fetch timeout stat increment which is not incrementing correctly now
Risk Level: Low
Testing: Updated
Docs Changes: N/A
Release Notes: N/A
Fixes https://github.com/envoyproxy/envoy/issues/15085

### CODE REVIEW

**Review by dmitri-d — APPROVED** [review]

**Review by junr03 — APPROVED** [review]

File: test/common/config/subscription_impl_test.cc [hunk 1]
```diff
@@ -179,9 +179,9 @@ TEST_P(SubscriptionImplInitFetchTimeoutTest, InitialFetchTimeout) {
   expectEnableInitFetchTimeoutTimer(std::chrono::milliseconds(1000));
   startSubscription({"cluster0", "cluster1"});
   EXPECT_TRUE(statsAre(1, 0, 0, 0, 0, 0, 0, ""));
-  if (GetParam() == SubscriptionType::Http) {
-    expectDisableInitFetchTimeoutTimer();
-  }
+
+  expectDisableInitFetchTimeoutTimer();
+
```
Comment (htuch) [inline test/common/config/subscription_impl_test.cc:184]: Can you add a test that validates the stats are updated?

Comment (ramaraochavali) [reply test/common/config/subscription_impl_test.cc:184]: It is already verified at the end of the test   EXPECT_TRUE(statsAre(1, 0, 0, 0, 1, 0, 0, ""));

Comment (htuch) [reply test/common/config/subscription_impl_test.cc:184]: Why was it passing before if that is validating the regression?

Comment (dmitri-d) [reply test/common/config/subscription_impl_test.cc:184]: see the implementation of ` expectConfigUpdateFailed()` before these changes (here: https://github.com/envoyproxy/envoy/pull/15114/files#diff-c09580e89a4c15ba79b2473fd1ae255b463fbde941649cb2420408b0261705a5L165).

Comment (htuch) [reply test/common/config/subscription_impl_test.cc:184]: Ah, I see. Thanks for pointing this out.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**